.github/workflows/go.yml

```diff
@@ -38,24 +38,25 @@ jobs:
         path: bin/*
 
   tests:
-    runs-on: macos-latest # Only Mac runners support nested virt
+    runs-on: ubuntu-latest # The runner must support nested virt
     needs: build # Don't bother testing if cross arch build fails
     timeout-minutes: 30
     steps:
     - uses: actions/checkout@v4
 
-    - name: Install
+    - name: Install qemu
       run: |
-        brew install qemu
-        touch continue
+        sudo apt install qemu-kvm
+        sudo usermod -a -G kvm $USER
 
     - name: Set up Go
       uses: actions/setup-go@v5
       with:
-        go-version: 1.20.x
+        go-version: stable
```

Comment:
The build is tested on 1.20/1.21/1.22, for running the tests, I don't have a strong preference between 1.20, 1.21 or 1.22. I used `stable` because of that, not really important.
An alternative could be to remove `with: go-version: ...` this is what is done when building releases.
Or we could also try https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file so that it always use the minimum go version we need.

Reply:
I moved this to a separate commit, it can be dropped if needed.

Reply:
I think https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file should be best option to make sure the test happens of the binary which built using same version of go which we also use.

Reply:
Done now.